rules/TypeDeclaration/Guard/PhpDocNestedAnnotationGuard.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\Guard;
+
+use Nette\Utils\Strings;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+final class PhpDocNestedAnnotationGuard
+{
+    /**
+     * Regex is used to count annotations including nested annotations
+     *
+     * @see https://regex101.com/r/G7wODT/1
+     */
+    private const SIMPLE_ANNOTATION_REGEX = '/@[A-z]+\(?/i';
+
+    private PhpDocInfoFactory $phpDocInfoFactory;
+
+    public function __construct(PhpDocInfoFactory $phpDocInfoFactory)
+    {
+        $this->phpDocInfoFactory = $phpDocInfoFactory;
+    }
+
+    /**
+     * Check if rector accidentally skipped annotation during parsing which it should not have (this bug is likely related to parsing of annotations
+     * in phpstan / rector)
+     */
+    public function isPhpDocCommentCorrectlyParsed(\PhpParser\Node $node): bool
```

Comment:
the usage is only for `Property` for this rule, so better to pass `PhpParser\Node\Stmt\Property` only:

```suggestion
    public function isPhpDocCommentCorrectlyParsed(\PhpParser\Node\Stmt\Property $node): bool
```

Reply:
I applied the change, but it will have to be changed back for the followup rector rules which also use this service.